.gitignore

```diff
@@ -4,4 +4,4 @@ cmake-build-debug/
 .vscode/
 README.html
 venv/
-
+*.swp
```

Comment:
Does everyone find this change acceptable and are there any conventions or style guides we are adhering to that include principals for `.gitignore`-ing editor temporary files? I would certainly find this change convenient, as I also use Vim, though I have heard some people object to including editor-specific temporary files in `.gitignore` files on the grounds of editor-agnosticism.

Reply:
IMO, if there is *any* file that may be created on your computer that should not be checked in, it's fair game to add it to gitignore. It doesn't hurt anybody, and saves us from accidentally committing files that shouldn't be there. 

Also, .vscode was already excluded, along with emacs backup files (*~)